patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,7 +9,27 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
-@@ -181,7 +181,7 @@
+                         break;
+                     }
+ 
+-                    tileentitybeacon$beamsegment.func_177262_a();
+-                    continue;
++                    // Allow any block to change the beacon beam color
++                    float[] color = iblockstate.func_177230_c().getBeaconColor(field_145850_b, iblockstate);
++                    if(color == null || color.length != 3){
++                        tileentitybeacon$beamsegment.func_177262_a();
```

Comment:
just use it directly (no checks), the runtime will do it for you :P 
It's a bit meh but we want to keep patch small